Hardware screening OS (#452)
home-assistant/architecture: adr/0017-hardware-screening-os.md

# 0017. Hardware screening for OS

> Date: 2021-06-30

Reviewer: ```suggestion
Date: 2021-08-03
```

## Status

> Draft

Reviewer: ```suggestion
Accepted
```

## Context

> We need take a decision for the minimal requirements of a IoT-Board to become a Home Assistant OS image.

Reviewer 2: Since every extra IoT board adds extra ongoing maintenance burden, is this a list of requirements that hardware needs to meet before it can be suggested? Or would any PR that adds support for hardware that meets these requirements be accepted?

Author: That is an interesting aspect. Just the machine build would be difficult at this point. They can run only generic qemu builds

Author: However, I will avoid the problem that we had with OrangePi. At least the required points need to exist that there is no maintenance burden for us.

Also how we deal with old or replaced generation/boards.

## Decision

Follow are the requirements of a Board:

- The board need to be available for everbody

> - The manifacture produce and sell the board at least 3 years

Reviewer 2: ```suggestion
- The manufacturer produces and sells the boards for at least 3 years
```

> - The manufacturer produces and sells the boards for at least 3 years

Reviewer: This will leave out RPi 4 and ODROID N2, those are not even 3y old.

Reviewer 3: I assume they mean that it will be produced for 3 years? So not a "special occasion board for the anniversary of something"-board.

Basically, most boards are considered outdated after 3 years...

Reviewer: What bord manufacture advertises how long a board will be produced?

Reviewer 3: Not many I guess, maybe @pvizeli or @balloob can explain what is meant with this requirement.

Author: Correct, there are many SBC manufacture which only sells a board 6-1y. That means, after we added the support, they are no longer available for new users. Special OrangePi is manufacture like this. Make all 3-6 months a new board. Since we need to buy new hardware for tests and don't make all the effort for something which is not available. RaspberryPi, ODROID is a manufacture which can be used also on B2B because these boards are a long time available and supported-

> - The manifacture is know for his continuity

Reviewer 4: ```suggestion
- The manufacturer is know for his continuity
```

> - The board need to be supported by mainstream Linux kernel with current or next LT

Reviewer 5: If it is supported by the upstream Kernel, it will come into the next long term anyways...

```suggestion
- The board needs to be supported by the upstream Linux kernel
```

Reviewer 5: Btw, maybe we need also mention what features should be supported at least. E.g. Ethernet, USB and SD-card...

> - The board need supported by u-boot/barebox bootloader

Reviewer 5: Is no upstream U-Boot/Barebox support required?

Author: it should, yes

> - The board needs to be supported by the upstream u-boot/barebox bootloader

Reviewer 5: ```suggestion
- The board needs to be supported by the upstream U-Boot or Barebox boot loader
```

> It's allow to use some patches as long they only minimal touch the Kernel and marked for upstream.

Reviewer 4: ```suggestion
It's allowed to use some patches as long they only minimal touch the Kernel and marked for upstream.
```

> We support the board as long as possible with updates so long as we don't have a downside and we have hardware

Reviewer 5: ```suggestion
We support the board as long as possible with updates. We might remove support if we see major downsides or we don't have hardware
```

> We support the board so long as possible with Updates as we don't have a downside and we have hardware
> for the sporadic tests. If there comes the next generation, we provide updates for the old but don't recommend it anymore.

Reviewer 4: ```suggestion
We support the board as long as possible with updates so long as we don't have a downside and we have hardware
for the sporadic tests.  When a next generation is available, we provide updates for the old but don't recommend it anymore.
```

> If a board is not supported anymore, they get no Updates from Operating-System or Home Assistant Core and according ADR0015.
> It's recommended to update the Hardware to next Generation or a new Board.

Reviewer 4: ```suggestion
If a board is no longer supported, no updates for the operating system or Home Assistant Core will be provided according ADR0015.
It's recommended to update the hardware to next generation or a new board.
```
